rawfile/utils/snapshot_manager.py

```diff
@@ -0,0 +1,115 @@
+from pathlib import Path
+import time
+from typing import List
+from dataclasses import dataclass
+import consts
+from utils.commands import run
+
+from utils.devices import device_to_mountpoint
+from utils.lock import VolLock
+from utils.rawfile import attached_loops, img_file, snapshots_dir
+from glob import glob
+
+
+@dataclass
+class Snapshot:
+    name: str
+    volume_id: str
+    size_bytes: int
+    creation_time: float
+    ready: bool = True
+
+
+class SnapshotManager:
+    def _get_snapshot_path(self, volume_id: str, name: str) -> Path:
+        return snapshots_dir(volume_id) / f"{name}.img"
+
+    def create_snapshot(
+        self,
+        volume_id: str,
+        name: str,
+        copy_on_write: bool = False,
+        freeze_fs: bool = False,
+    ) -> Snapshot:
+        with VolLock(volume_id):
+            file = img_file(volume_id)
+            loop_devs = attached_loops(file.as_posix())
+            snap_path = self._get_snapshot_path(volume_id, name)
+            snap_path.parent.mkdir(parents=True, exist_ok=True)
+            Path(f"{snap_path}.creating").touch()
```

Comment:
Should we add metadata files for snapshots as well rather than just looking at img files?

Reply:
By metadata files, we will look at img files later on too, since we need the exact size of the file